myriad-scheduler/src/main/java/com/ebay/myriad/Main.java

```diff
@@ -211,7 +215,32 @@ private void startNMInstances(Injector injector) {
       Map<String, Integer> nmInstances = injector.getInstance(MyriadConfiguration.class).getNmInstances();
       MyriadOperations myriadOperations = injector.getInstance(MyriadOperations.class);
       ServiceProfileManager profileManager = injector.getInstance(ServiceProfileManager.class);
+      SchedulerState schedulerState = injector.getInstance(SchedulerState.class);
+
+      Set<NodeTask> launchedNMTasks = new HashSet<>();
+      launchedNMTasks.addAll(
+        schedulerState.getPendingTasksByType(NodeManagerConfiguration.NM_TASK_PREFIX));
+      if (!launchedNMTasks.isEmpty()) {
```

Comment:
Currently I am checking if there is at least one NM in pending, staging or active state on restart (irrespective of profile type). If this is the first time the cluster is being launched, there will be no NM running. In this case we look at the myriad-default-config.xml to look at number of NMs to be launched per profile.